finalPrep/modZeroHero.js

```diff
@@ -55,30 +79,74 @@ var excuse = 'I think I forgot to lock up my 1992 Toyota Coralla. Be right back.
 //   - citiesDestroyed (array)
 //   - luckyNumbers (array)
 //   - address (object with following key/values: number , street , state, zip)
-
+var scaryMonster = {
+  name = "Teke Teke",
+  smell = "Death",
+  weight = 3,
+  citiesDestroyed = ["London", "Paris", "New York"],
+  luckyNumbers = [3, 7, 13],
+  address: {
+    number: 13,
+    street: "Evil Street",
+    state: "FL",
+    zip:30315};
+};
 
 // Create a new class called SuperHero
 // - Your class should have the following DYNAMIC values
-//   - name 
+//   - name
 //   - superpower
-//   - age 
+//   - age
 // - Your class should have the following STATIC values
 //   - archNemesis, assigned to "The Syntax Error"
 //   - powerLevel = 100
-//   - energyLevel = 50 
+//   - energyLevel = 50
+class SuperHero {
+  constructor(name, superpower, age) {
+  this.name = name;
+  this.superpower = superpower;
+  this.age = age;
+  this.archNemesis = "The Syntax Error";
+  this.powerLevel = 100;
+  this.energyLevel = 50;
+}
 
 // - Create the following class methods
 //   - sayName, should print the hero's name to the console
 //   - maximizeEnergy, should update the energyLevel to 1000
 //   - gainPower, should take an argument of a number and INCREASE the powerLevel by that number
 
+sayName(nameNew) {
+  this.name = nameNew;
+}
+maximizeEnergy(energyLevelNew) {
+  this.energyLevel = 1000;
+}
+gainPower(powerLevelNew) {
+  this.powerLevel = this.powerLevel + powerLevelNew;
+}
+
+};
 // - Create 2 instances of your SuperHero class
+var superHero1 = new SuperHero ("Meta Man", "Weather Control", 20);
+//console.log(superHero1);
+//var superHero2 = new SuperHero ("Captain Granite", "Fire Breathing", 65, "The Syntax Error", 100, 50);
+//console.log(superHero2);
+superHero1.sayName("Bob The Rooster"); superHero1.maximizeEnergy(); superHero1.gainPower(50);
+console.log(superHero1);
 
+//superHero2.sayName("Random");superHero2.maximizeEnergy();
+//superHero2.gainPower();
+//console.log(superHero2);
 
 // Reflection
 // What parts were most difficult about this exerise?
 
+//The Function part... I struggle a bit with the relationshiup between creating
+//variable and applying them to Fucntions. Where, when, how... Need to practice.
+
 // What parts felt most comfortable to you?
+// The early part, creating variables, etc...
 
 // What skills do you need to continue to practice before starting Mod 1?
-
+//Functions, classes, class methods...
```

Comment:
You'll definitely get more practice with these concepts in Mod 1. You don't have to 100% understand this stuff yet.